sdk/samples/router/general.py

```diff
@@ -0,0 +1,308 @@
+        enricher=enricher,
+        ensembler=ensembler
+    )
+
+    # 1. Create a new router using the RouterConfig object
+    new_router = turing.Router.create(router_config)
+
+    # 2. List all routers
+    routers = turing.Router.list()
+    for r in routers:
+        if r.name == new_router.name:
+            my_router = r
+        print(r)
+
+    # Wait for the router to get deployed
+    try:
+        my_router.wait_for_status(RouterStatus.DEPLOYED)
+    except TimeoutError:
+        raise Exception(f"Turing API is taking too long for router {my_router.id} to get deployed.")
+
+    # 3. Get the router you just created using the router_id obtained
+    my_router = turing.Router.get(my_router.id)
+
+    # Access the router config from the returned Router object directly
+    my_router_config = my_router.config
+
+    # Modify something in the router config
+    my_router_config.routes.append(
+        Route(
+            id='fee-fi-fo-fum',
+            endpoint='http://fox-says.fee-fi-fo-fum',
+            timeout='20ms'
+        )
+    )
+
+    # 4. Update the router with the new router config
+    my_router.update(my_router_config)
+
+    # 5. List all the router config versions of your router
+    my_router_versions = my_router.list_versions()
+    for ver in my_router_versions:
+        print(ver)
+
+    # Sort the versions returned by version number
+    my_router_versions.sort(key=lambda x: x.version)
+    # Get the version number of the first version returned
+    first_ver_no = my_router_versions[0].version
+    # Get the version number of the latest version returned
+    latest_ver_no = my_router_versions[-1].version
+
+    # Wait for the first version to get undeployed
+    try:
+        my_router.wait_for_version_status(RouterStatus.UNDEPLOYED, first_ver_no)
+    except TimeoutError:
+        raise Exception(f"Turing API is taking too long for router {my_router.id} with version {first_ver_no} to get "
+                        f"undeployed.")
+
+    # 6. Deploy a specific router config version (the first one we created)
+    response = my_router.deploy_version(first_ver_no)
+    print(f"6. You have deployed version {response['version']} of router {response['router_id']}.")
+
+    # Wait for the first version to get deployed
+    try:
+        my_router.wait_for_version_status(RouterStatus.DEPLOYED, first_ver_no)
+    except TimeoutError:
+        raise Exception(f"Turing API is taking too long for router {my_router.id} with version {first_ver_no} to get "
+                        f"deployed.")
+
+    # 7. Undeploy the current active router configuration
+    response = my_router.undeploy()
+    print(f"7. You have undeployed router {response['router_id']}.")
+
+    # Wait for the router to get undeployed
+    try:
+        my_router.wait_for_status(RouterStatus.UNDEPLOYED)
+    except TimeoutError:
+        raise Exception(f"Turing API is taking too long for router {my_router.id} to get undeployed.")
+
+    time.sleep(30)
```

Comment:
This call of `sleep` is a little awkward; it's there to somehow prevent this odd case where the deployment of a router fails when it is undeployed and immediately redeployed.